indico/modules/events/timetable/client/js/DayTimetable.tsx

```diff
@@ -200,7 +203,22 @@ export function DayTimetable({dt, eventId, minHour, maxHour, entries}: DayTimeta
     if (!newLayout) {
       return;
     }
-    dispatch(actions.moveEntry(dt.format('YYYYMMDD'), newLayout));
+    let movedEntry = newLayout.find(e => e.id === who);
+
+    if (!movedEntry) {
+      for (const block of newLayout) {
+        if (block.children) {
+          movedEntry = block.children.find(child => child.id === who);
+          if (movedEntry) {
+            break;
+          }
+        }
+      }
+    }
```

Comment:
Instead of searching for the entry again (which could be costly for a large number of entries), we could return the new entry from `layoutAfterDropOnBlock` and `layoutAfterDropOnCalendar`. It could look like this:
```ts
const [newLayout, movedEntry] = layoutAfterDropOnBlock(...);
if (!newLayout) {
  return;
}
dispatch(actions.moveEntry(movedEntry, ...)));
```